api/add.go

```diff
@@ -0,0 +1,115 @@
+package api
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/weaveworks/pctl/pkg/bootstrap"
+	"github.com/weaveworks/pctl/pkg/catalog"
+	"github.com/weaveworks/pctl/pkg/git"
+	"github.com/weaveworks/pctl/pkg/install"
+	"github.com/weaveworks/pctl/pkg/log"
+)
+
+// AddProfileOpts defines settings for adding profiles.
+type AddProfileOpts struct {
+	URL           string
+	Branch        string
+	SubName       string
+	Namespace     string
+	ConfigMap     string
+	Dir           string
+	Path          string
+	GitRepository string
```

Comment:
Same here, could be namespace and name separately

Reply:
Or even better `NamespacedName` with better typing!